validation/scripts/test_discovery.py

```diff
@@ -39,33 +42,65 @@ def get_test_data():
 
     for test in tests:
         # layers/data tuples
-        layers, data = get_data(test)
+        layers_file, data_file = get_data(test)
 
-        slabs = np.loadtxt(layers)
+        slabs = np.loadtxt(os.path.join(PTH, layers_file))
         assert slabs.shape[1] == 4
 
-        data = np.loadtxt(data)
+        data = np.loadtxt(os.path.join(PTH, data_file))
         assert data.shape[1] in [2, 3, 4]
 
         yield os.path.basename(test), slabs, data
 
 
 def get_data(file):
-    # for each of the unpolarised test files figure out where the data and
-    # layer parameters are
-
-    with open(file, "r") as f:
-        # ignore comment lines starting with # or space
-        while True:
-            line = f.readline()
-            if line.lstrip(" \t").startswith("#"):
-                continue
-            elif line == "\n":
-                continue
-            else:
-                layers = line.rstrip("\n")
-                data = f.readline().rstrip("\n")
-
-                layers = os.path.join(PTH, layers)
-                data = os.path.join(PTH, data)
-                return (layers, data)
+    # for each of the test files extract the parameters,
+    # e.g. the names of the data_file and layers_file
+
+    with open(file, "rt") as f:
+        # ignore comment lines starting with #
+        for line in f:
+            line = line.strip()
+            if not line.startswith("#"):
+                yield line
+
+
+def get_polarised_test_data():
+    """
+    A generator yielding (slabs, data) tuples.
+
+    `slabs` are np.ndarrays that specify the layer structure of the test.
+    ``slabs.shape = (N + 2, 6)``, where N is the number of layers.
+
+    The layer specification file has the layout:
+
+    # { "AGUIDE": angle_between_Q_and_H }
```

Comment:
This says that AGUIDE is contained in the *.layers files, but they're actually contained in the description file.

Reply:
How many packages deal with experiments for which AGUIDE is not 90 or 270? Is it up to the individual validations to skip those tests if that's the case?

Reply:
Yes, I would prefer to leave in the tests for AGUIDE with an out of plane component (not exactly 90 or 270, as you say) and then skip them for programs that don't support that capability.

And that is a good catch on the documentation - as I was writing these tests I originally intended to put H and AGUIDE in the layers file but I later realized there is more flexibility to reuse layer files with different values of AGUIDE and H if it is in the description instead, but I never updated the docs.

Reply:
See [9f9101c](https://github.com/reflectivity/analysis/pull/66/commits/9f9101c753d8b4b2e9ff4386e6a9b9873d82d147)